packages/react/src/components/ContentSwitcher/ContentSwitcher.js

```diff
@@ -63,18 +77,43 @@ export default class ContentSwitcher extends React.Component {
         ]),
         onKeyDown: this.handleChildChange,
         selected: index === this.state.selectedIndex,
+        ref: switchRefs[index],
       })
     );
   }
 
   handleChildChange = data => {
+    // the currently selected child index
     const { selectedIndex } = this.state;
+    // the newly selected child index
     const { index } = data;
+    const { key } = data;
 
-    if (selectedIndex !== index) {
-      this.setState({ selectedIndex: index });
-      this.props.onChange(data);
+    if (key) {
+      const nextIndex = getNextIndex(
+        key,
+        selectedIndex,
+        this.props.children.length
+      );
+      this.setState(
+        {
+          selectedIndex: nextIndex,
+        },
+        () => {
+          const switchRef = this._switchRefs[nextIndex];
+          switchRef && switchRef.focus();
+        }
+      );
+    } else {
+      if (selectedIndex !== index) {
+        this.setState({ selectedIndex: index }, () => {
+          const switchRef = this._switchRefs[index];
+          switchRef && switchRef.focus();
+        });
+      }
     }
+
+    this.props.onChange(data);
```

Comment:
Should this be called in the `setState` callbacks above after the transition occurs?